digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/strategy/TrackCalculatorStrategy.scala

```diff
@@ -179,11 +179,13 @@ trait TrackCalculatorStrategy {
     def fixedMinimimumAddress = Math.max(Math.max(rightProjectLinks.last.startAddrMValue + 1, leftProjectLinks.last.startAddrMValue + 1), fixedAddress)
     def addressLengthRight = Math.max(0, rightProjectLinks.last.originalEndAddrMValue - rightProjectLinks.last.originalStartAddrMValue)
     def addressLengthLeft  = Math.max(0, leftProjectLinks.last.originalEndAddrMValue  -  leftProjectLinks.last.originalStartAddrMValue)
+
     val minimumEndAddress = (leftProjectLinks.exists(_.status == LinkStatus.New), rightProjectLinks.exists(_.status == LinkStatus.New)) match {
       case (true,true) => fixedMinimimumAddress
       case (true, false)  => rightProjectLinks.last.startAddrMValue + addressLengthRight
       case (false, true)  => leftProjectLinks.last.startAddrMValue + addressLengthLeft
-      case (false,false)  => fixedMinimimumAddress
+      case (false,false)  => Math.floor((startSectionAddress + leftProjectLinks.last.endAddrMValue - leftProjectLinks.head.startAddrMValue +
+                                         startSectionAddress + rightProjectLinks.last.endAddrMValue - rightProjectLinks.head.startAddrMValue) * 0.5).round
```

Comment:
"/2" would more explicitly refer to averaging, than "0.5".
-> use /2 instead of 0.5, if there are no logical reasons to use a decimal number

Reply:
(...also, refactoring the averaging to a small function might be a good thing)

Reply:
Ha! Käytetty jo olemassa olevaa funktiota! Vielä parempi fiksi! <3
(Tosin, sielläkin näyttää käytetyn 0.5:sta. Noh... No niin. :D)